src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.cs

```diff
@@ -302,15 +302,22 @@ public string FriendlyName
         {
             get
             {
-                int cbData = 0;
-                if (!Interop.crypt32.CertGetCertificateContextPropertyString(_certContext, CertContextPropId.CERT_FRIENDLY_NAME_PROP_ID, null, ref cbData))
-                    return string.Empty;
+                unsafe
+                {
+                    int cbData = 0;
+                    if (!Interop.crypt32.CertGetCertificateContextPropertyString(_certContext, CertContextPropId.CERT_FRIENDLY_NAME_PROP_ID, null, ref cbData))
+                        return string.Empty;
 
-                StringBuilder sb = new StringBuilder((cbData + 1) / 2);
-                if (!Interop.crypt32.CertGetCertificateContextPropertyString(_certContext, CertContextPropId.CERT_FRIENDLY_NAME_PROP_ID, sb, ref cbData))
-                    return string.Empty;
+                    int spanLength = (cbData + 1) / 2;
+                    Span<char> buffer = spanLength <= 128 ? stackalloc char[spanLength] : new char[spanLength];
```

Comment:
Would it make sense to use `ArrayPool` when `spanLength > 128`?
It makes the code a little bit clumsly for keeping track of the array from the pool, but it avoids an allocation. 
Or is in this interop-case the advantage from the ArrayPool negligible?

Same at the other places where this pattern is used.

Reply:
My understanding of these use cases is that these strings are generally relatively short, and given that, it didn't seem worth the complexity for the corner case where we'd need to heap allocate.  But I could be wrong.  @bartonjs?

Reply:
128 is pretty reasonable for friendly name -- it's usually a domain name or company name and those aren't usually that long.

Reply:
That said, it's also not that common to use friendly name on a hot path (for security, you're supposed to use Subject), so I'm not sure what the need for this change is.

Reply:
>  so I'm not sure what the need for this change is.

To get rid of all StringBuilder marshaling in corefx.  It's like the difference between having 0 warnings and 1 warning; the moment you have any, it's a slippery slope to easily have more, whereas it's easy to stay clean when there are 0.

Reply:
> for security, you're supposed to use Subject), so I'm not sure what the need for this change is.

This PR also fixes this for Subject.

Reply:
Ok, I'll bump it to 256. Thanks.